modules/auxiliary/gather/zookeeper_info_disclosure.rb

```
@@ -0,0 +1,80 @@
+          ['URL', 'https://zookeeper.apache.org/doc/current/zookeeperAdmin.html']
+        ],
+      'Author' =>
+        [
+          'Karn Ganeshen <KarnGaneshen[at]gmail.com>'
+        ],
+      'DisclosureDate' => 'Oct 14, 2020',
+      'License' => MSF_LICENSE,
+      'DefaultOptions' => { 'VERBOSE' => true })
+      )
+
+    register_options(
+      [
+        Opt::RPORT(2181),
+        OptInt.new('TIMEOUT', [true, 'Timeout for the probe', 30])
+      ], self.class
+    )
+
+    deregister_options('USERNAME', 'PASSWORD')
+  end
+
+  def run_host(ip)
+    to = (datastore['TIMEOUT'].zero?) ? 30 : datastore['TIMEOUT']
+    begin
+      ::Timeout.timeout(to) do
+        connect
+        print_status('Dumping environment info...')
+        sock.put('environ')
+        data = sock.get_once(-1, to).to_s
+        print_good("#{data}")
+        sock.close
+
+        loot_name = 'environ-log'
+        loot_type = 'text/plain'
+        loot_desc = 'Zookeeper Environment Log'
+        loot_service = 'Zookeeper'
+        p = store_loot(loot_name, loot_type, datastore['RHOST'], data, loot_desc, loot_service)
+        print_good("File saved in: #{p} \n")
+        report_service(host: rhost, port: rport, name: 'Zookeeper', info: 'Apache Zookeeper')
+
+        connect
+        print_status('Dumping statistics about performance and connected clients...')
+        sock.put('stat')
+        data = sock.get_once(-1, to).to_s
+        print_good("#{data}")
+        sock.close
+
+        loot_name = 'stat-log'
+        loot_type = 'text/plain'
+        loot_desc = 'Zookeeper Stat Log'
+        loot_service = 'Zookeeper'
+        p = store_loot(loot_name, loot_type, datastore['RHOST'], data, loot_desc, loot_service)
+        print_good("File saved in: #{p} \n")
+        report_service(host: rhost, port: rport, name: 'Zookeeper', info: 'Apache Zookeeper')
+      end
+    rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError
```

The code that runs `run_host` also silently catches timeouts, which in this context can be confusing, so adding a `rescue Timeout::Error, Rex::TimeoutError` clause here would help the user understand when a host could not be scanned/dumped because it did not respond to the request.

---

Changed to `rescue Timeout::Error, ::Rex::TimeoutError, ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError`